samtranslator/model/sam_resources.py

```diff
@@ -1224,6 +1225,15 @@ def _get_retention_policy_value(self):
         :return: value for the DeletionPolicy attribute.
         """
 
+        if is_intrinsic(self.RetentionPolicy):
+            # RetentionPolicy attribute of AWS::Serverless::LayerVersion does set the DeletionPolicy
+            # attribute. And DeletionPolicy attribute does not support intrinsic values.
+            raise InvalidResourceException(
```

Comment:
Should we communicate that the customer could use the top level Attribute of DeletionPolicy? For customers needing or wanting the Intrinsic, this might be the only way forward for them.

Reply:
Unfortunately, intrinsics are not supported for top level attributes; https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html

I tested with a template but I wasn't able to deploy it.